netket_tools/python_utils.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
+# Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Netket python utilities.
+"""
+import json
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def encode_complex(z):
```

Comment:
Is this used to encode complex hamiltonians, or not yet?

Reply:
One can use this function when they dump out all the parameters (including hamiltonians) and this will handle any complex values, e.g.
```python
json.dump(pars, outfile, default=encode_complex)
```
This will correctly encode `pars['Hamiltonian']` if it contains any complex values.

Reply:
ok great! Then maybe we can even hide this call to the dump method into some help function called ```GenerateInputFile`` of something similar/with a better name !